### PR TITLE
normalize gitconfig's author field to UTF-8 NFC

### PR DESCRIPTION
**Pull request by jmg-duarte**
# Pull Request Check List

Resolves: #8779

- [x] Added **tests** for changed code. 
- [ ] Updated **documentation** for changed code.

Not sure if the checklist is applicable to this one, let me know!

### CODE REVIEW

**Comment by radoering** [comment]
Can you add a test, please?
(In poetry-core, there is already a test for `combine_unicode` so I suppose it should be possible to add a test for `init`, here.)

**Comment by jmg-duarte** [comment]
> Can you add a test, please?

Added. Had to make `_validate_author` static, but it's inline with other validations.

I'm currently just testing for the offending character though, let me know if you know of more troublemakers I should add!

File: src/poetry/console/commands/init.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def handle(self) -> int:
         question = self.create_question(
             f"Author [<comment>{author}</comment>, n to skip]: ", default=author
         )
-        question.set_validator(lambda v: self._validate_author(v, author))
+        question.set_validator(lambda v: InitCommand._validate_author(v, author))
```
Comment (radoering) [inline src/poetry/console/commands/init.py:145]: No need to change this line. (We can access a static method via `self`.)

File: tests/console/commands/test_init.py [hunk 2]
```diff
@@ -993,6 +993,24 @@ def test_validate_package_invalid(name: str) -> None:
         assert InitCommand._validate_package(name)
 
 
+@pytest.mark.parametrize(
+    "author",
+    [
+        # As MacOS saves "é" — 65 CC 81
+        "José Duarte",
+        # As written by a MacOS keyboard — C3 A9
+        "José Duarte",
```
Comment (radoering) [inline tests/console/commands/test_init.py:1002]: It seems like both strings are the same (C3 A9). The test succeeds even after reverting your change.

**Review by radoering — CHANGES_REQUESTED** [review]

File: src/poetry/console/commands/init.py (hunk 1 above)
Comment (jmg-duarte) [reply src/poetry/console/commands/init.py:145]: TIL

**Review by radoering — APPROVED** [review]

**Pull request merged**